### DISCUSSION TITLE
Is there a way to allow expired presigned GET url (for integration testing)?

### DISCUSSION THREAD

**Discussion by dystopiandev** [minio/minio#20089] [Q&A] [answered]
Localstack allows this by default, since it's a dev tool. So I'm wondering if it's possible to configure minio to work the same way.

**Comment by klauspost** [comment, marked as answer]
No. We are not interested in functionality like this. First of all, it is a risk factor to have code in there that could potentially be bypassed either by a configuration mistake or a coding mistake, or even as a secondary attack vector. Second of all having significant portions of your code take different paths devalues the tests.

Other people can do tests without this. I suggest you look into alternatives where you do the signing live.

**Reply by dystopiandev** [reply]
It was the one test that checks the validity of the `Expires` query param by freezing the time provider before generating the URL. I could of course use the current time and check for a close enough timestamp, but I prefer the exact check.
That said, I've worked around it already. Thanks.
